ui/src/track-layout/layout-map-api.ts

```diff
@@ -102,38 +102,73 @@ export async function getMapAlignmentsByTiles(
     changeTimes: ChangeTimes,
     mapTiles: MapTile[],
     publishType: PublishType,
-    fetchType: AlignmentFetchType,
-    selectedId?: AlignmentId,
-): Promise<AlignmentDataHolder[]> {
-    const alignmentChanged = getMaxTimestamp(
-        changeTimes.layoutLocationTrack,
-        changeTimes.layoutReferenceLine,
-    );
-    const rlChanged = getMaxTimestamp(
-        changeTimes.layoutTrackNumber,
-        changeTimes.layoutReferenceLine,
+) {
+    const polyLines = await Promise.all(
+        mapTiles.map((tile) =>
+            getPolyLines(tile, changeTimes.layoutReferenceLine, publishType, 'ALL'),
+        ),
+    ).then((p) => p.flat());
+
+    const [rlPolyLines, ltPolyLines] = partitionBy(
+        polyLines,
+        (i) => i.alignmentType === 'REFERENCE_LINE',
     );
-    const ltChanged = changeTimes.layoutLocationTrack;
-    const polyLines = (
-        await Promise.all(
-            mapTiles.map((tile) =>
-                getPolyLines(tile, alignmentChanged, publishType, fetchType, selectedId),
-            ),
-        )
-    ).flat();
-    const rlIds = getIds(polyLines, 'REFERENCE_LINE');
-    const ltIds = getIds(polyLines, 'LOCATION_TRACK');
-    const rlHeaders = await getAlignmentHeaders(publishType, rlIds, 'REFERENCE_LINE', rlChanged);
-    const ltHeaders = await getAlignmentHeaders(publishType, ltIds, 'LOCATION_TRACK', ltChanged);
-    const trackNumbers = await getTrackNumbers(publishType, changeTimes.layoutTrackNumber);
-    return combine([...rlHeaders, ...ltHeaders], polyLines, trackNumbers);
+
+    return [
+        ...(await getAlignmentDataHolder('REFERENCE_LINE', rlPolyLines, publishType, changeTimes)),
+        ...(await getAlignmentDataHolder('LOCATION_TRACK', ltPolyLines, publishType, changeTimes)),
+    ];
+}
+
+export async function getReferenceLineMapAlignmentsByTiles(
+    changeTimes: ChangeTimes,
+    mapTiles: MapTile[],
+    publishType: PublishType,
+) {
+    const polyLines = await Promise.all(
+        mapTiles.map((tile) =>
+            getPolyLines(tile, changeTimes.layoutReferenceLine, publishType, 'REFERENCE_LINES'),
+        ),
+    ).then((p) => p.flat());
+
+    return getAlignmentDataHolder('REFERENCE_LINE', polyLines, publishType, changeTimes);
+}
+
+export async function getLocationTrackMapAlignmentsByTiles(
+    changeTimes: ChangeTimes,
+    mapTiles: MapTile[],
+    publishType: PublishType,
+    locationTrackId?: LocationTrackId,
+) {
+    const polyLines = await Promise.all(
+        mapTiles.map((tile) =>
+            getPolyLines(tile, changeTimes.layoutLocationTrack, publishType, 'LOCATION_TRACKS'),
+        ),
+    )
+        .then((lines) => lines.flat())
+        .then((lines) => {
+            return lines.filter((line) => (locationTrackId ? line.id === locationTrackId : true));
```

Comment:
Onkos tässä nyt pelkona, että kun zoomataan ulos joku raide valittuna, niin käydään hakemassa koko Suomen kaikki locationTrackit? 🤔

Reply:
On joo vaarana, jäin hieman kahen vaiheille kummin tuon nyt toteuttaisi.
Pyörittelin kans ajatusta, jos ois uus rajapinta jolla vois hakea MapAlignmentin pelkällä ID:llä, mutta se bounding box on leivottu sen verran syvälle, ett en ruvennut purkaa sitä. Ja taas yhden raiteen hakeminen bounding boxilla on hieman weird.

Reply:
Ei se ehkä niin kovin weird ole. Pitkät raiteet on tosi pitkiä ne ne polyline pitää joka tapauksessa myös yksinkertaistaa, valittu tai ei, niin miksipä samalla myös poimia vain siltä kohtaa kun UI:ta oikeasti kiinnostaa. Tämä on tosiaan vähän huono että haetaan iso kasa raiteita ilman että niitä halutaan edes näyttää.

Reply:
Tässä muutoksessahan on siis se hyvä puoli että tuo valinta ei nyt myöskään myrkytä tuota cachea kun selectedId ei ole enää cache-avaimessa mukana. Eli kaikkia raiteita ei tarvitse hakea uudelleen aina kun valitaan eri raide :+1: 

Tuon voisi tosin hoitaa myös niin että tuo suodatus tehdään backarissa (optionaalinen id filtteröinti arg backarissa, muuten sama toiminta kuin nytkin) ja täällä haetaan filtterin kera eri cachen läpi (tai vain cachen ohi)